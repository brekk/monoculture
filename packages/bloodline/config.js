export const CONFIG = {
  alias: {
    help: ['h'],
    color: ['k'],
    input: ['i'],
    output: ['o'],
    basePath: ['b'],
  },
  configuration: {
    'strip-aliased': true,
  },
}

export const CONFIG_DEFAULTS = {
  color: true,
}

/* eslint-disable max-len */
export const HELP_CONFIG = {
  help: `This text you're reading now!`,
  color: `Do stuff with glorious color`,
  input: `Where to start building a bloodline`,
  output: `Where to write an output file`,
  basePath: `Process dependencies relative to this path.`,
}
/* eslint-enable max-len */
