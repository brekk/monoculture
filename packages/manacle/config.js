export const CONFIG = {
  alias: {
    help: ['h'],
    color: ['k'],
    input: ['i'],
    query: ['q'],
    check: ['c'],
    filter: ['f'],
  },
  boolean: ['help', 'color'],
  array: ['check', 'filter'],
  configuration: {
    'strip-aliased': true,
  },
}

export const CONFIG_DEFAULTS = {
  color: true,
  help: false,
}

/* eslint-disable max-len */
export const HELP_CONFIG = {
  help: `This text you're reading now!`,
  color: `Do stuff with glorious color`,
  input: `The path to a monocle-findings.json file.`,
  query: `Ask a question of the data`,
  check: `Provide a checkfile (Can be specified multiple times)`,
  filter: `Filter the results and cut down the output`,
}
/* eslint-enable max-len */

export const BANNER = `     _     _
    ( \\._./ )
    _)  ●  (_
   /    ⟁    \\
  /  .-----.  \\
 /  /       \\  \\
(  (         )  )
 \\  \\_______/  /
  '-___|_____-'
`
