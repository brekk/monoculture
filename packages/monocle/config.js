export const CONFIG = {
  alias: {
    help: ['h'],
    showMatchesOnly: ['m', 'showMatches'],
    showTotalMatchesOnly: ['n', 'showTotalMatches'],
    rulefile: ['c'],
    ignore: ['i'],
    plugins: ['p', 'plugin'],
    rule: ['r', 'rules'],
    error: ['e'],
    trim: ['t'],
    output: ['o'],
    jsonIndent: ['j'],
    color: ['k'],
  },
  number: ['jsonIndent'],
  boolean: ['help', 'trim', 'showMatchesOnly', 'color'],
  array: ['plugins', 'rule', 'ignore'],
  configuration: {
    'strip-aliased': true,
  },
}

export const CONFIG_DEFAULTS = {
  output: 'monocle-findings.json',
  showMatchesOnly: false,
  showTotalMatchesOnly: false,
  jsonIndent: 2,
  color: true,
}

/* eslint-disable max-len */
export const HELP_CONFIG = {
  help: `This text you're reading now!`,
  color: `Do stuff with glorious color`,
  ignore: `Pass ignore values to glob. Array type`,
  trim: 'Trim the lines on read',
  plugins: `Specify plugin(s) to add to the run.
  Multiple plugins can be specified by invoking monocle with multiple flags, e.g.
    monocle --plugin x --plugin y`,
  rule: `Specify a rule to add to the run.
  Multiple rules can be specified by invoking monocle with multiple flags, e.g.
    monocle --rule one --rule two`,
  error: 'Should this invocation be a warning or an error?',
  rulefile: `Express the context of a run in a single file.
  This overrides all other flags (except --help)`,
  output: 'Where should the findings be written?',
  showMatchesOnly: `List the files which matched the search, but *do not* run plugins`,
  showTotalMatchesOnly: `List the number files which matched the search, but *do not* run plugins (implies -m)`,
  jsonIndent: `How much should we indent JSON output? (default: 2)`,
}
/* eslint-enable max-len */
