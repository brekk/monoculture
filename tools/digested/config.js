export const YARGS_CONFIG = {
  alias: {
    // boolean
    color: ['k'],
    readme: ['m'],
    deps: ['s'],
    api: ['s'],
    // string
    banner: ['b'],
    bannerPath: ['B'],
    pkgPath: ['pkg', 'i'],
    drGenPath: ['drgen', 'd'],
    docUrl: ['u'],
    repoUrl: ['r'],
  },
  boolean: ['readme', 'deps', 'api'],
  configuration: {
    'strip-aliased': true,
  },
}

export const HELP_CONFIG = {
  banner:
    'Provide a string to insert before the rendered content. (only applicable with `--readme`)',
  bannerPath:
    // eslint-disable-next-line max-len
    'Provide a path to a file to insert before the rendered content. (only applicable with `--readme`)',
  pkgPath: 'The path to the `package.json` file.',
  drGenPath: 'An optional path to a dr-generated.json file',
  docUrl: 'The URI of a documentation site, used in generating links',
  repoUrl: 'The URI of a repository, used in generating links',
  // booleans
  help: 'This text!',
  color: 'Render stuff in glorious color',
  readme: 'Generate content for a readme (in markdown!)',
  deps: 'Add dependency information [default: true] (only applicable with `--readme`)',
  api: 'Add API information [default: true] (only applicable with `--readme`)',
}

export const CONFIG_DEFAULTS = {
  readme: false,
  api: true,
  deps: true,
  color: true,
}
