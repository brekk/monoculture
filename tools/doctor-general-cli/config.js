import { curry } from 'ramda'
import yargsParser from 'yargs-parser'

// parser :: YargsConfig -> List String -> Object
export const parser = curry(function _parser(opts, args) {
  return yargsParser(args, opts)
})

export const YARGS_CONFIG = {
  alias: {
    input: ['i'],
    output: ['o'],
    search: ['s'],
    artifact: ['a'],
    ignore: ['g'],
    color: ['k'],
    debug: ['d'],
    monorepo: ['m'],
    processor: ['p'],
  },
  array: ['input'],
  boolean: ['color', 'debug', 'monorepo'],
  configuration: {
    'strip-aliased': true,
  },
}

export const HELP_CONFIG = {
  help: 'This text!',
  color: 'Render stuff in color',
  input:
    // eslint-disable-next-line max-len
    'A file to read! (Can be specified multiple times.) If running in monorepo mode, this should be your root `package.json`.',
  output: 'The file to output!',
  search: "The glob to use for searching (default: '**/*.{js,jsx,ts,tsx}')",
  artifact: `Would you like to create an artifact file?
(Useful for downstream transformation)`,
  ignore: 'Files to ignore when searching, can be specified multiple times',
  debug: 'Generate additional information when processing content.',
  monorepo: 'Process content for a monorepo (walking all "workspaces")',
  processor: 'Specify a processor to use when running doctor-general',
}

export const CONFIG_DEFAULTS = {
  color: true,
  ignore: [
    '**/node_modules/**',
    '**/coverage/**',
    '**/*.spec.{js,jsx,ts,tsx}',
    '**/fixture/**',
    '**/fixture.*',
  ],
  search: '**/*.{js,jsx,ts,tsx}',
  debug: false,
}
