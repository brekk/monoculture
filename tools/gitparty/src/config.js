/* eslint-disable max-len */

export const YARGS_CONFIG = {
  configuration: {
    'strip-aliased': true,
  },
  alias: {
    help: ['h'],
    init: ['i'],
    color: ['k'],
    config: ['c'],
    repo: ['r'],
    timezone: ['t', 'tz'],
    filter: ['f'],
    collapseAuthors: ['a', 'collapse'],
    json: ['j'],
    excludeMergeCommits: ['m'],
    output: ['o'],
    totalCommits: ['n'],
    aliases: ['a', 'alias'],
    dateFormat: ['d'],
    width: ['w'],
    fields: ['e'],
  },
  boolean: ['excludeMergeCommits', 'collapseAuthors', 'init'],
  array: ['aliases', 'fields'],
  number: ['width'],
}
export const CONFIG_DEFAULTS = {
  // this is our config
  excludeMergeCommits: true,
  collapseAuthors: false,
  filter: ``,
  json: false,
  // fill in at runtime
  // repo: process.cwd(),
  // below this line are gitlog configuration
  totalCommits: 50,
  fields: [
    'abbrevHash',
    'subject',
    'authorName',
    'authorDate',
    'authorDateRel',
  ],
  timezone: `UTC`,
  execOptions: { maxBuffer: 1000 * 1024 },
  color: true,
  dateFormat: 'HH:mm',
}
export const HELP_CONFIG = {
  help: 'This text!',
  color: 'Render things in glorious color!',
  init: 'Initialize and create a new `.gitpartyrc` file',
  config: 'Select a config file other than `.gitpartyrc`',
  output:
    'Write the results to a file. Usually useful in concert with the --json flag',
  json: 'Return JSON ouput instead of rendering as strings',
  repo: 'Select a different git repository',
  totalCommits: 'Choose the number of total commits',
  collapseAuthors:
    'Merge commits if the authors are the same and commit dates are the same',
  excludeMergeCommits: 'Exclude merge commits from the results',
  filter: c => {
    const { yellow: y, bold: d, cyan: n } = c
    return `Filter commits based on a simple '${y('key')}:${n('value')}' / '${y(
      'key'
    )}:${n('value')}#${y('key2')}:${n('value2')}' base syntax:
\`-f "${y('hash')}:${n('80ca7f7')}" / -f "${y('date')}:${n('20-05-2018')}"\`
\tLookup by exact string matching (default)
\`-f "${y('subject')}:${n('fix~')}"\`
\tLookup by looser indexOf matching when there is a tilde "~" character at the end of the value
\`-f "${y('subject')}:${n('fix~')}#${y('date')}:${n('20-05-2018')}"\`
\tLookup with multiple facets, separated by a hash "#" symbol
\`-f "${y('author')}:${n('brekk')}"\`
\tWhen filtering by author, the alias lookup (if aliases have been defined) is used
\`-f "${y('files')}:${n('**/src/*.spec.js')}"\`
\tWhen there are asterisks present in the value side (after the ":") and the key is an array, glob-style matching is performed
\`-f "${y('x')}:${n('true')}" / -f "${y('x')}:${n('false')}"\`
\tWhen the value is either the literal string "true" or "false", it will be coerced into a boolean
\`-f "${y('analysis.config')}:${n('true')}"\`
\tWhen there is a period "." in the key, nested-key lookups will be performed`
  },
  timezone: 'Set the timezone you want to see results in. Defaults to UTC',
  aliases:
    'Define author aliases (useful if authors do not merge under consistent git `user.name`s / `user.email`s)',
  dateFormat:
    "Express how you want date-fns to format your dates. (default: 'yyyy-MM-dd HH:kk OOOO')",
  width: 'Set an explicit width',
  fields: 'Specify the fields you want to pull from `gitlog`',
}
export const DEFAULT_CONFIG_FILE = {
  patterns: {
    gitpartyrc: {
      key: 'G',
      color: 'bgRed',
      matches: ['**/.gitpartyrc*', '**/.gitpartyrc'],
    },
  },
  collapseAuthors: false,
  timezone: 'UTC',
}
