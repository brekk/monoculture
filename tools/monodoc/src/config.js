import { curry } from 'ramda'
import yargsParser from 'yargs-parser'

// parser :: YargsConfig -> List String -> Object
export const parser = curry((opts, args) => yargsParser(args, opts))

export const YARGS_CONFIG = {
  alias: {
    input: ['i'],
    output: ['o'],
    search: ['s'],
    artifact: ['a'],
    ignore: ['g'],
    color: ['k'],
  },
  boolean: ['color'],
  configuration: {
    'strip-aliased': true,
  },
}

export const HELP_CONFIG = {
  help: 'This text!',
  color: 'Render stuff in color',
  input: 'A file to read!',
  output: 'The file to output!',
  search: "The glob to use for searching (default: '**//*.{js,jsx,ts,tsx}')",
  artifact: `Would you like to create an artifact file?
(Useful for downstream transformation)`,
  ignore: 'Files to ignore when searching, can be specified multiple times',
}

export const CONFIG_DEFAULTS = {
  color: true,
  ignore: [
    '**/node_modules/**',
    '**/coverage/**',
    '**/*.spec.{js,jsx,ts,tsx}',
    '**/fixture/**',
    '**/fixture.*',
  ],
  search: '**/*.{js,jsx,ts,tsx}',
}
