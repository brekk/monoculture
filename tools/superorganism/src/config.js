export const YARGS_CONFIG = {
  alias: {
    // these come from `nps`
    silent: ['s'],
    scripts: [],
    config: ['c'],
    logLevel: ['l'],
    require: ['r'],
    helpStyle: ['y'],
    // these are things we've added
    future: ['f'],
    color: ['k'],
  },
  array: ['require'],
  boolean: ['silent', 'scripts', 'future', 'color'],
  configuration: {
    'strip-aliased': true,
  },
}

/* eslint-disable max-len */
export const HELP_CONFIG = {
  help: 'This text!',
  silent: 'Silence superorganism output',
  scripts: 'Log command text for script',
  config: `Config file to use (defaults to nearest package-scripts.js)`,
  logLevel: `The log level to use (error | warn | info | debug)`,
  require: `Module to preload`,
  helpStyle: `Choose the level of detail displayed by the help command`,
  future: `Use Futures instead of Promises`,
  color: `Render things with color`,
}
/* eslint-enable max-len */

export const CONFIG_DEFAULTS = {
  scripts: true,
  future: false,
  helpStyle: 'all',
  color: true,
}
