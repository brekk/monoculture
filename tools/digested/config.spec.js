import { YARGS_CONFIG, HELP_CONFIG, CONFIG_DEFAULTS } from './config'

test('YARGS_CONFIG', () => {
  expect(YARGS_CONFIG).toMatchSnapshot()
})
test('HELP_CONFIG', () => {
  expect(HELP_CONFIG).toMatchSnapshot()
})
test('CONFIG_DEFAULTS', () => {
  expect(CONFIG_DEFAULTS).toMatchSnapshot()
})
