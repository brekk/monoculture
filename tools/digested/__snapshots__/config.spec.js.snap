// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CONFIG_DEFAULTS 1`] = `
{
  "api": true,
  "color": true,
  "deps": true,
  "readme": false,
}
`;

exports[`HELP_CONFIG 1`] = `
{
  "api": "Add API information [default: true] (only applicable with \`--readme\`)",
  "banner": "Provide a string to insert before the rendered content. (only applicable with \`--readme\`)",
  "bannerPath": "Provide a path to a file to insert before the rendered content. (only applicable with \`--readme\`)",
  "color": "Render stuff in glorious color",
  "deps": "Add dependency information [default: true] (only applicable with \`--readme\`)",
  "docUrl": "The URI of a documentation site, used in generating links",
  "drGenPath": "An optional path to a dr-generated.json file",
  "help": "This text!",
  "pkgPath": "The path to the \`package.json\` file.",
  "readme": "Generate content for a readme (in markdown!)",
  "repoUrl": "The URI of a repository, used in generating links",
}
`;

exports[`YARGS_CONFIG 1`] = `
{
  "alias": {
    "api": [
      "s",
    ],
    "banner": [
      "b",
    ],
    "bannerPath": [
      "B",
    ],
    "color": [
      "k",
    ],
    "deps": [
      "s",
    ],
    "docUrl": [
      "u",
    ],
    "drGenPath": [
      "drgen",
      "d",
    ],
    "pkgPath": [
      "pkg",
      "i",
    ],
    "readme": [
      "m",
    ],
    "repoUrl": [
      "r",
    ],
  },
  "boolean": [
    "readme",
    "deps",
    "api",
  ],
  "configuration": {
    "strip-aliased": true,
  },
}
`;
